### PR TITLE
feat(markdown): support hasPragma/insertPragma

### PR DESCRIPTION
**Pull request by ikatyang**
_hasPragma_

**support**

```md
<!-- @prettier -->
```

```md
---
something
---

<!-- @prettier -->
```

```md
<!--
  MIT LICENSE 
  
  bla bla bla

  @format
-->
```

---

_insertPragma_

**support**

```diff
+ <!-- @prettier -->
+
something something
```

```diff
---
seomthing
---

+ <!-- @prettier -->
+
something something
```

**does not support**

```diff
<!--
  MIT LICENSE 
  
  bla bla bla
+
+ @format
-->
```

### CODE REVIEW

File: src/language-markdown/pragma.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+const pragmas = ["format", "prettier"];
+const startWithPragmaRegex = new RegExp(
+  `^<!--\\s*@(${pragmas.join("|")})\\s*-->`
+);
+
+module.exports = {
+  startWithPragmaRegex,
+  hasPragma: text => startWithPragmaRegex.test(text.trimLeft()),
```
Comment (suchipi) [inline src/language-markdown/pragma.js]: Do we need to worry about this case?

Comment (suchipi) [reply src/language-markdown/pragma.js]: Or:
```md
<!--
  MIT LICENSE

  bla bla bla
-->

<!-- @format -->
```

Comment (ikatyang) [reply src/language-markdown/pragma.js]: We should worry about the first one, but I'm not sure about the second one, maybe just doing something like JS?
```md
<!--
  MIT LICENSE 
  
  bla bla bla

  @format
-->
```

Comment (suchipi) [reply src/language-markdown/pragma.js]: That looks good to me :+1:

File: tests/insert-pragma/markdown/simple.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+
```
Comment (j-f1) [inline tests/insert-pragma/markdown/simple.md]: Can you add a test without these blank lines?

**Comment by ikatyang** [comment]
Don't know how to handle this case so I think we can just apply the common rule for now.

```diff
+<!-- @format -->

 <!--
   some
     thing
    not 
      indented
 well
 -->
```

**Pull request merged**